python/lsst/cp/pipe/defects.py

```diff
@@ -367,27 +393,91 @@ def _writeData(self, dataRef, defects):
         defects : `lsst.meas.algorithms.Defect`
             The defects to be written.
         """
-        filename = dataRef.getUri(write=True)  # does not guarantee that full path exists
-        dirname = os.path.dirname(filename)
-        if not os.path.exists(dirname):
-            os.makedirs(dirname)
+        date = str(dafBase.DateTime(midTime, dafBase.DateTime.MJD).toPython())
```

Comment:
does:
```python
date = dafBase.DateTime(midTime, dafBase.DateTime.MJD).toPython().isoformat()
```
work so you don't need the next line?

Reply:
I think so - Chris suggested the same above. Will check and change if so.